runtime/lua/vim/loader.lua

```diff
@@ -0,0 +1,525 @@
+local uv = vim.loop
+
+local M = {}
+
+---@alias CacheHash {mtime: {sec:number, nsec:number}, size:number}
+---@alias CacheEntry {hash:CacheHash, chunk:string}
+
+---@class ModuleFindOpts
+---@field all? boolean Search for all matches (defaults to `false`)
+---@field rtp? boolean Search for modname in the runtime path (defaults to `true`)
+---@field patterns? string[] Paterns to use (defaults to `{"/init.lua", ".lua"}`)
```

Comment:
Typo here. Also does "patterns" here mean "Lua patterns"? Given that Vim has its own "pattern" concept, it might be good to be explicit.

```suggestion
---@field patterns? string[] Lua patterns to use (defaults to `{"/init.lua", ".lua"}`)
```

Reply:
Not lua patterns and also not vim patterns.

Reply:
So glob patterns then? Or just plain strings? The term "patterns" is ambiguous in Vim precisely because of how many different pattern "schemes" are in use.

Reply:
no, just plain strings.

Reply:
In the Lua context, though, it's usually (and should always be) used for "Lua patterns" (aka "we already have regex at home").

Reply:
Those aren't lua patterns. They're just strings that are added to the basename of the lua module.

Reply:
What should I rename it to? Any suggestions?

Reply:
The name itself is fine to me, as long as the docstring is explicit about the semantics.

>They're just strings that are added to the basename of the lua module.

This is useful information to include in the docstring. As a user of this API this is exactly what I would want to know.